service/awsconfig/framework.go

```diff
@@ -451,19 +445,92 @@ func newVersionedResources(config FrameworkConfig) (map[string][]framework.Resou
 		}
 	}
 
-	// We provide a map of resource lists keyed by the version bundle version
-	// to the resource router.
-	versionedResources := map[string][]framework.Resource{
-		// Clusters without a version use the legacy resource.
-		"":      legacyResources,
-		"0.1.0": legacyResources,
-		"0.2.0": resourcesV2,
-		"1.0.0": legacyResources,
-		"2.0.0": resourcesV2,
-		"2.0.1": resourcesV2_0_1,
-		// 2.0.2 fixes missing region in host account credentials, the change only affects service/framework.go
-		"2.0.2": resourcesV2_0_1,
+	handlesFuncFunc := func(handledVersionBundles []string) func(obj interface{}) bool {
+		return func(obj interface{}) bool {
+			kvmConfig, err := key.ToCustomObject(obj)
+			if err != nil {
+				return false
+			}
+			versionBundleVersion := key.VersionBundleVersion(kvmConfig)
+
+			for _, v := range handledVersionBundles {
+				if versionBundleVersion == v {
+					return true
+				}
+			}
+
+			return false
+		}
+	}
+
+	var resourceSetV1 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"",
+				"0.1.0",
+				"1.0.0",
+			}),
+			Logger:    config.Logger,
+			Resources: legacyResources,
+		}
+
+		resourceSetV1, err = framework.NewResourceSet(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourceSetV2 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"0.2.0",
+				"2.0.0",
+			}),
+			Logger:    config.Logger,
+			Resources: resourcesV2,
+		}
+
+		resourceSetV2, err = framework.NewResourceSet(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourceSetV4 *framework.ResourceSet
+	{
+		c := framework.ResourceSetConfig{
+			Handles: handlesFuncFunc([]string{
+				"2.0.1",
+				// 2.0.2 fixes missing region in host account credentials, the change only affects service/framework.go
+				"2.0.2",
```

Comment:
BTW I don't understand the purpose of 2.0.2. It was bug fix in the operator. I'd stay we shouldn't introduce a new version for that.

Reply:
You're right `2.0.2` was a bug fix to the operator to add the missing host region flag. Adding a new version confused things.